### PR TITLE
Issue #927: Suppress silent-no-op false positive for clean review phase

### LINKED ISSUE #927: detect-wrapper-anomaly: review phase の clean review 完了時に silent-no-op を誤検出
## Background

Issue #917 の `/auto` 実行中 (review phase, PR #926) で、`scripts/detect-wrapper-anomaly.sh` の `silent-no-op` (#365) 判定が誤検出した:

```
[#917] [anomaly] silent no-op detected in review:
### Orchestration Anomalies
- **[silent-no-op]** LLM reported success in phase `review` (exit code 0) but no commit for #926 found in recent git log. Possible silent no-op: ...

### Improvement Proposals
- Re-run `run-code.sh 926` to retry the code phase. ...
```

しかし実際には `gh pr view 926` で PR は MERGED、"## Acceptance Criteria Verification Results" を含む Review も正常投稿済みだった。MUST 修正が不要な clean review だったため新規コミットが存在しなかっただけで、review phase は完全に成功していた。

`run-auto-sub.sh` はこの anomaly を情報ログとしてのみ扱う設計 (`EXIT_CODE=0` 分岐で無条件に `return 0`) のため pipeline 自体への実害はなかったが、以下のリスクがある:

- Spec の Auto Retrospective / `docs/reports/orchestration-recoveries.md` に事実誤認レコードが残る
- 将来 `recoveries-auto-fire` 系の自動集計が発火した場合、無害な clean review を誤って recovery 対象として扱うリスク

これは #916 が `merge` phase 限定で修正した「`gh pr view` による live 状態チェックで false positive を抑止する」パターンと同型だが、`review` phase には未適用。

## Purpose

`scripts/detect-wrapper-anomaly.sh` の `silent-no-op` 判定を `review` phase についても live-state チェックで補正し、clean review (MUST 修正不要、コミットなしで完了) を誤って anomaly として報告しないようにする。

## Proposal (Outline)

`review` phase かつ `EXIT_CODE=0` かつ完了フレーズ検出時、コミット有無だけで anomaly 判定する前に `gh pr view <PR> --json reviews` で "Acceptance Criteria Verification Results" を含む Review が既に投稿されているかを確認する。投稿済みなら (新規コミット有無に関わらず) anomaly を抑止する。#916 で `merge` phase に導入した `gh pr view --json state` の MERGED live check と同型のパターン。

### Pre-merge (auto-verified)

- [x] `detect-wrapper-anomaly.sh` の review phase silent-no-op 検出が、"Acceptance Criteria Verification Results" を含む Review が既に投稿されている場合には anomaly を報告しないよう修正されている 
- [x]  上記の live チェックが `scripts/detect-wrapper-anomaly.sh` 内に実装されている
- [x] bats test で、review phase かつ clean review (Review 投稿済み・新規コミットなし) のケースに対して anomaly が発生しないことが検証されている 

### Post-merge

- [x] 次回 `/auto` の pr route review phase で clean review が完了した際、false-positive silent-no-op が発生しないことを観察 

## Auto-Resolved Ambiguity Points

- **[`gh pr view --json reviews` 呼び出し自体が失敗した場合 (API エラー・rate limit 等) は、review 投稿未確認として扱い anomaly 抑止を行わない (fail-safe)]** — reason: #916 が同型の `MERGED` live check で採用した「確認不能時は既存の silent-no-op 判定ロジックにフォールスルーする (fail-open ではなく fail-safe な安全側デフォルト)」という先例判断と一致させた。加えて `scripts/post-fallback-review-summary.sh` (同種の "Acceptance Criteria Verification Results" 検出ロジックを既に実装済み) も `gh pr view --json reviews` 失敗時 (`|| true`) を「未検出」として扱っており、既存コードベースパターンとも整合する。
  - Other candidates: 呼び出し失敗時に anomaly を無条件で抑止する (fail-open) — false positive は減るが、真の silent-no-op を見逃すリスクが増えるため不採用。

## Related

- 発生元: Issue #917 の `/auto` 実行中 (PR #926 review phase)
- 対象 file: `scripts/detect-wrapper-anomaly.sh`
- 関連: #916 (merge phase の同型 false positive 修正)

### PR DESCRIPTION
**Pull request by saito**
## Summary
`scripts/detect-wrapper-anomaly.sh` の `silent-no-op` 判定に `review` phase 専用の live チェックを追加。`gh pr view <PR> --json reviews --jq '.reviews[].body'` を実行し、出力に "Acceptance Criteria Verification Results" が含まれる場合は (新規コミット有無に関わらず) silent-no-op anomaly の報告を抑止する。#916 で `merge` phase に導入した `gh pr view --json state` の MERGED live check と同型のパターン。`gh pr view` 呼び出し自体が失敗した場合は fail-safe (既存ロジックへフォールスルー、抑止しない)。

## Verification (pre-merge)
- `detect-wrapper-anomaly.sh` の review phase silent-no-op 検出が、"Acceptance Criteria Verification Results" を含む Review が既に投稿されている場合には anomaly を報告しないよう修正されている
- 上記の live チェックが `scripts/detect-wrapper-anomaly.sh` 内に実装されている
- bats test で、review phase かつ clean review (Review 投稿済み・新規コミットなし) のケースに対して anomaly が発生しないことが検証されている (`tests/detect-wrapper-anomaly.bats` に3ケース追加、既存分と合わせて全1094件PASS)

## Verification (post-merge)
- 次回 `/auto` の pr route review phase で clean review が完了した際、false-positive silent-no-op が発生しないことを観察 (opportunistic)

closes #927

### CODE REVIEW

File: scripts/detect-wrapper-anomaly.sh [hunk 1]
```diff
@@ -102,8 +102,17 @@ elif [[ "$EXIT_CODE" == "0" ]]; then
       _merge_pr_confirmed_merged=true
     fi
   fi
+  _review_confirmed_posted=false
+  if [[ "$PHASE" == "review" ]]; then
+    _review_bodies=$(gh pr view "$ISSUE_NUMBER" --json reviews --jq '.reviews[].body' 2>/dev/null)
```
Comment (saito) [inline scripts/detect-wrapper-anomaly.sh:107]: **[Spec Deviation] SHOULD**
The design rationale (`modules/orchestration-fallbacks.md`) frames this live check as detecting "the clean review case (MUST fixes not required, no new commit produced)". But `## Acceptance Criteria Verification Results` is the standard heading of every posted Review body — clean or with unresolved MUST rows alike. The implementation only greps for the heading string; it never checks whether the matched review is actually free of MUST issues. So the stated design intent is narrower than what the code actually verifies.

Recommended fix: either narrow the check to also require absence of MUST-severity content, or correct the doc wording to describe the actual, broader condition ("any prior Review containing this heading exists").

Comment (saito) [inline scripts/detect-wrapper-anomaly.sh:107]: **[Edge Case/Robustness] SHOULD**
`gh pr view --json reviews --jq '.reviews[].body'` returns the concatenation of all Reviews ever posted on the PR, not just the current invocation's output. Since the review phase can legitimately be retried after a kill/API-error (per `orchestration-fallbacks.md`'s documented recovery paths), a PR can accumulate a qualifying Review from an earlier attempt. On a later retry where the session genuinely silent-no-ops, the stale Review from the earlier attempt would still satisfy the grep and permanently suppress the anomaly.

Note: `scripts/post-fallback-review-summary.sh` uses the identical unscoped full-history pattern, so this is a pre-existing accepted tradeoff in the codebase rather than a regression introduced by this PR. Flagging for awareness — no fix required in this PR's scope; consider a follow-up Issue if this proves to matter in practice.

File: tests/detect-wrapper-anomaly.bats [hunk 2]
```diff
@@ -173,6 +173,67 @@ MOCK
     [[ "$output" == *"### Orchestration Anomalies"* ]]
 }
 
+@test "silent no-op: suppressed for review phase when gh pr view confirms Review posted" {
```
Comment (saito) [inline tests/detect-wrapper-anomaly.bats:176]: **[Edge Case/Robustness] SHOULD**
The added test suite covers isolated scenarios (Review posted / no matching Review / `gh` failure) but no regression test for the stale/history-based false-suppression scenario noted above. Since that scoping behavior matches an existing codebase pattern (`post-fallback-review-summary.sh`) rather than being introduced by this PR, this is optional follow-up test coverage rather than a blocking gap.

File: modules/orchestration-fallbacks.md [hunk 3]
```diff
@@ -480,6 +480,8 @@ When `reconcile-phase-state.sh --check-completion` returns `"matches_expected":t
...
 
+For the `review` phase, `detect-wrapper-anomaly.sh` similarly runs an independent live check (`gh pr view <PR> --json reviews --jq '.reviews[].body'`) before falling through to the log/git-log-based checks above. If the output contains "Acceptance Criteria Verification Results", the silent-no-op entry is skipped regardless of wrapper log content or local git history — this covers the clean review case (MUST fixes not required, no new commit produced) where the review itself completed successfully. If the `gh pr view` call fails or no matching Review is found, detection falls through to the existing logic (fail-safe, not fail-open; Issue #927).
```
Comment (saito) [inline modules/orchestration-fallbacks.md]: **[Documentation Consistency] SHOULD**
The added sentence states the live check "covers the clean review case (MUST fixes not required, no new commit produced)" — this over-specifies the actual guard, which matches any Review containing the heading regardless of MUST status or which phase invocation produced it. Recommend rephrasing to state the actual condition precisely, e.g.: "If any Review (from the current or a prior cycle) contains this heading, the silent-no-op entry is skipped — this does not verify absence of unresolved MUST issues."

**Review by saito — COMMENTED** [review]
## Acceptance Criteria Verification Results

| Condition | Result | Notes |
|-----------|--------|-------|
| detect-wrapper-anomaly.sh の review phase silent-no-op 検出が、Review 投稿済みの場合 anomaly を報告しないよう修正されている | PASS | rubric判定: fail-safe な live check ロジックが実装コードで確認できた |
| 上記の live チェックが scripts/detect-wrapper-anomaly.sh 内に実装されている (file_contains) | PASS | `"Acceptance Criteria Verification Results"` 文字列の grep を確認 |
| bats test で clean review ケースの anomaly 抑止が検証されている | PASS | rubric判定: 3ケース (抑止/未マッチ検出/gh失敗フォールスルー) がbatsに追加されていることを確認 |
| (Post-merge) 次回 `/auto` の pr route review phase での opportunistic 観察 | POST-MERGE | マージ後に自然発生を待つ |

## CI Status

| Job name | Status | Notes |
|---------|--------|-------|
| DCO | SUCCESS | — |
| Run bats tests | SUCCESS | — |
| Validate skill syntax | SUCCESS | — |
| Forbidden Expressions check | SUCCESS | — |
| macOS shell compatibility | SUCCESS | — |

## Code Review

### General Comments (issues where path/line cannot be identified)

None.

### Issue Count Summary

- Posted as line comments: 4 (MUST: 0, SHOULD: 4, CONSIDER: 0)
- General Comments (in body): 0

---
Next: run `/merge 931`.

**Comment by saito** [comment]
## Review Response Summary

This is an auto-generated fallback summary, posted by `post-fallback-review-summary.sh` after the review session exited without posting its own Response Summary (silent no-op). A prior review with Acceptance Criteria Verification Results was found for this PR.

Before merging, manually confirm CI status and the content of any fix commits pushed during this review.
